src/lib/storage/dictionary_segment.cpp

```diff
@@ -30,76 +31,76 @@ template <typename T>
     : BaseDictionarySegment(data_type_from_type<T>()) {
-  if constexpr (std::is_same_v<T, int32_t>) {
-    const auto encoding_type = PersistedSegmentEncodingType{start_address[ENCODING_TYPE_OFFSET_INDEX]};
-    const auto dictionary_size = start_address[DICTIONARY_SIZE_OFFSET_INDEX];
-    const auto attribute_vector_size = start_address[ATTRIBUTE_VECTOR_OFFSET_INDEX];
-
-    auto* dictionary_address = reinterpret_cast<const T*>(start_address + HEADER_OFFSET_INDEX);
-    auto dictionary_span_pointer = std::make_shared<std::span<const T>>(dictionary_address, dictionary_size);
-
-    switch (encoding_type) {
-      case PersistedSegmentEncodingType::DictionaryEncoding8Bit: {
-        auto* const attribute_vector_address =
-            reinterpret_cast<const uint8_t*>(start_address + HEADER_OFFSET_INDEX + dictionary_size);
-        auto attribute_data_span = std::span<const uint8_t>(attribute_vector_address, attribute_vector_size);
-        auto attribute_vector = std::make_shared<FixedWidthIntegerVector<uint8_t>>(attribute_data_span);
-
-        _dictionary_span = dictionary_span_pointer;
-        _attribute_vector = attribute_vector;
-        _decompressor = _attribute_vector->create_base_decompressor();
-
-        break;
-      }
-      case PersistedSegmentEncodingType::DictionaryEncoding16Bit: {
-        auto* const attribute_vector_address =
-            reinterpret_cast<const uint16_t*>(start_address + HEADER_OFFSET_INDEX + dictionary_size);
-        auto attribute_data_span = std::span<const uint16_t>(attribute_vector_address, attribute_vector_size);
-        auto attribute_vector = std::make_shared<FixedWidthIntegerVector<uint16_t>>(attribute_data_span);
-
-        _dictionary_span = dictionary_span_pointer;
-        _attribute_vector = attribute_vector;
-        _decompressor = _attribute_vector->create_base_decompressor();
-
-        break;
-      }
-      case PersistedSegmentEncodingType::DictionaryEncoding32Bit: {
-        auto* const attribute_vector_address =
-          reinterpret_cast<const uint32_t*>(start_address + HEADER_OFFSET_INDEX + dictionary_size);
-        auto attribute_data_span = std::span<const uint32_t>(attribute_vector_address, attribute_vector_size);
-        auto attribute_vector = std::make_shared<FixedWidthIntegerVector<uint32_t>>(attribute_data_span);
-
-        _dictionary_span = dictionary_span_pointer;
-        _attribute_vector = attribute_vector;
-        _decompressor = _attribute_vector->create_base_decompressor();
-
-        break;
-      }
-      case PersistedSegmentEncodingType::DictionaryEncodingBitPacking: {
-        Fail("Support for span-based BitPackingVectors for DictionarySegments not implemented yet.");
-        break;
-      }
-      default: {
-        Fail("Unsupported EncodingType.");
-        break;
-      }
+  const auto header_data = reinterpret_cast<const uint32_t*>(start_address);
+  const auto encoding_type = PersistedSegmentEncodingType{header_data[ENCODING_TYPE_OFFSET_INDEX]};
```

Comment:
Do we need this additional enum? All it stores has been available in  existing enums before?